packages/gestalt/src/Popover/Contents.js

```diff
@@ -0,0 +1,133 @@
+// @flow strict
+import { type Node, useEffect, useRef } from 'react';
+import { FloatingFocusManager } from '@floating-ui/react';
+import classnames from 'classnames';
+import usePopover, { DIRECTIONS_MAP, SIDES_MAP } from './usePopover.js';
+import borders from '../Borders.css';
+import Caret from '../Caret.js';
+import colors from '../Colors.css';
+import styles from '../Contents.css';
+import { type MainDirections } from '../utils/positioningTypes.js';
+import { CARET_HEIGHT, CARET_WIDTH } from '../utils/positioningUtils.js';
+
+export type Role = 'dialog' | 'listbox' | 'menu' | 'tooltip';
+
+type Props = {
+  accessibilityLabel?: string,
+  anchor: HTMLElement,
+  bgColor: 'blue' | 'darkGray' | 'orange' | 'red' | 'white',
+  border?: boolean,
+  caret?: boolean,
+  children?: Node,
+  id: ?string,
+  idealDirection?: MainDirections,
+  onKeyDown: (event: SyntheticKeyboardEvent<HTMLElement>) => void,
+  role: ?Role,
+  rounding?: 2 | 4,
+  shouldFocus?: boolean,
+  width: ?number,
+  scrollBoundary?: HTMLElement,
+};
+
+export default function Contents({
+  accessibilityLabel,
+  anchor,
+  bgColor,
+  border = true,
+  caret = true,
+  children,
+  id,
+  idealDirection,
+  role,
+  rounding,
+  width,
+  shouldFocus = true,
+  onKeyDown,
+  scrollBoundary,
+}: Props): Node {
+  const caretRef = useRef<HTMLElement | null>(null);
+  const idealPlacement = idealDirection ? DIRECTIONS_MAP[idealDirection] : 'top';
+
+  const { refs, placement, floatingStyles, middlewareData, context } = usePopover({
+    anchor,
+    caretElement: caretRef.current,
+    direction: idealPlacement,
+    scrollBoundary,
+  });
+
+  const caretOffset = middlewareData.arrow;
+  const visibility = middlewareData.hide?.referenceHidden === true ? 'hidden' : 'visible';
+  const background = bgColor === 'white' ? `${bgColor}BgElevated` : `${bgColor}Bg`;
+  const bgColorElevated = bgColor === 'white' ? 'whiteElevated' : bgColor;
+  const isCaretVertical = placement === 'top' || placement === 'bottom';
+
+  const popoverCaret = (
```

Comment:
nit?
any benefit of keeping it here?
could we just inline this? more readable IMO inlined